src/gui/painting/qpainter.cpp

```diff
@@ -2673,6 +2673,152 @@ QPainterPath QPainter::clipPath() const
     }
 }
 
+static QPainterPath noClipping(const QTransform &matrix, const QPainterClipInfo &info)
+{
+    return QPainterPath();
+}
+
+static QPainterPath PathClipReplaceClip(const QTransform &matrix, const QPainterClipInfo &info)
+{
+    return info.path * matrix;
+}
+
+static QPainterPath PathClipIntersectClip(const QTransform &matrix, const QPainterClipInfo &info)
+{
+    return PathClipReplaceClip(matrix, info);
+}
+
+static QPainterPath RectClipReplaceClip(const QTransform &matrix, const QPainterClipInfo &info)
+{
+    QPainterPath tempPath;
+    tempPath.addRect(info.rect);
+    tempPath = tempPath * matrix;
+    return tempPath;
+}
+
+static QPainterPath RectClipIntersectClip(const QTransform &matrix, const QPainterClipInfo &info)
+{
+    return RectClipReplaceClip(matrix, info);
+}
+
+static QPainterPath RectFClipReplaceClip(const QTransform &matrix, const QPainterClipInfo &info)
+{
+    QPainterPath tempPath;
+    tempPath.addRect(info.rectf);
+    tempPath = tempPath * matrix;
+    return tempPath;
+}
+
+static QPainterPath RectFClipIntersectClip(const QTransform &matrix, const QPainterClipInfo &info)
```

Comment:
Can you help me understand why this is the only one that is quite different between the intersect and replace versions?

Seems like you could just always do tempPath.addRect(info.rectf) and multiply by matrix afterwards (which would obviously make this the same as the replace version)

Reply:
Still applies - I don't understand 🤕